tools/python/runfiles/runfiles.py

```diff
@@ -161,6 +176,52 @@ def EnvVars(self):
     """
     return self._strategy.EnvVars()
 
+  def CurrentRepository(self, frame = 1):
+    """Returns the canonical name of the caller's Bazel repository.
+
+    For example, this function returns '' (the empty string) when called from
+    the main repository and a string of the form 'rules_python~0.13.0` when
+    called from code in the repository corresponding to the rules_python Bazel
+    module.
+
+    More information about the difference between canonical repository names and
+    the `@repo` part of labels is available at:
+    https://bazel.build/build/bzlmod#repository-names
+
+    NOTE: This function inspects the callstack to determine where in the
+    runfiles the caller is located to determine which repository it came from.
+    This may fail or produce incorrect results depending on who the caller is.
+    Use the `frame` argument to control the stack lookup, or pre-resolve the
+    repository via other means, for example by passing it in from Starlark, to
```

Comment:
"by passing it in from starlark" -- how is one supposed to do this?

Reply:
One could pass in `Label.workspace_name` (sic, this should really be called `repository_name`) via an argument or environment variable.

I would rather have people
1. try calling this function from somewhere else if they are doing something "interesting" on the Python side.
2. file an issue so we learn about this.

Would you be fine me changing this comment to say that?

Reply:
Do you happen to have an example of a situation in which you would expect this to fail? If we can't think of any, we should maybe tell users to file issues directly.

Reply:
To clarify, I'm not asking for any changes here. I was just asking a question. I wasn't aware of Label.workspace_name previously.

The main thing I'm after here is that the doc clearly says it has known limitations and what those limitations are. You shouldn't have to do a deep dive of source code to diagnose an error when you're providing apparently valid inputs, only to find out your inputs aren't _actually_ valid but fall into some known but undocumented edge case.

I'm fine with the two doc changes you suggested.

> Do you happen to have an example of a situation in which you would expect this to fail?

The main case that comes to mind is whenever inspect.getfile() would fail, likely due to getframe() ending up in some builtin (i.e. C) function somehow.

Reply:
I added a note saying that this could for example fail if the caller isn't represented by a Python source file and removed the part about Starlark. Rather than passing in repository names, in edge cases it would be better to pass in entire runfiles paths. I will make sure to document that possibility when I update the `Rlocation` function.